plaid-stl/src/github/mod.rs

```diff
@@ -29,6 +29,99 @@ pub struct RepositoryCollaborator {
     pub permissions: Permission,
 }
 
+// Structs used for deserializing GH's responses when searching for code
+
+#[derive(Debug, Deserialize)]
+pub struct FileSearchResult {
+    pub total_count: u64,
+    pub incomplete_results: bool,
+    pub items: Vec<FileSearchResultItem>,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct FileSearchResultItem {
+    pub path: String,
+    pub html_url: String,
+    pub url: String,
+    pub repository: GithubRepository,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct GithubRepository {
+    pub name: String,
+    pub full_name: String,
+    pub private: bool,
+    pub description: Option<String>,
+    pub owner: GithubRepositoryOwner,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct GithubRepositoryOwner {
+    pub login: String,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct GithubFileContent {
+    #[serde(rename = "type")]
+    pub type_: String,
+    pub content: String,
+    pub encoding: String,
+}
+
+// END OF Structs used for deserializing GH's response when searching for code
+
+impl FileSearchResultItem {
+    /// Retrieve the content of the search result
+    pub fn retrieve_raw_content(&self) -> Result<String, PlaidFunctionError> {
+        let reference_regex = regex::Regex::new(r"^.*?ref=([a-f0-9]{40})$").unwrap(); // TODO improve
```

Comment:
This is going to make this call very expensive because compilation of regexes is intensive. If there's no way around it then it's fine (regex doesn't support static regexes yet) but it might be something we need to watch for when running simple rules using it because they could run out of compute :p